releasePipeline/05-deploy-maven-galasa.sh

```diff
@@ -0,0 +1,208 @@
+
+    read -p "Please enter the version number: " version
+
+    echo "You are doing a release for version ${version}"
+}
+
+
+function set_kubernetes_context {
+    namespace="galasa-build"
+    h1 "Setting the kubernetes context to be cicsk8s, using namespace ${namespace}"
+    kubectl config set-context cicsk8s --namespace=${namespace}
+    rc=$?
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed. rc=${rc}"
+        exit 1
+    fi
+    success "Kubernetes context set to cicsk8s using namespace ${namespace}"
+}
+
+
+
+function deploy_maven_artifacts {
+
+    h1 "Deploying all artifacts in github called ${release_type}"
+
+    branch_name="${release_type}"
+
+    rm -f temp/deploy-maven-galasa.yaml
+    cat << EOF > temp/deploy-maven-galasa.yaml
+
+#
+# Copyright contributors to the Galasa project
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+kind: PipelineRun
+apiVersion: tekton.dev/v1beta1
+metadata:
+  generateName: deploy-maven-galasa-
+  annotations:
+    argocd.argoproj.io/compare-options: IgnoreExtraneous
+    argocd.argoproj.io/sync-options: Prune=false
+spec:
+  params:
+  - name: version
+    value: "{$version}"
+  - name: image
+    value: harbor.galasa.dev/galasadev/galasa-obr-with-galasabld:"{$release_type}"
+  pipelineRef:
+    name: deploy-maven-galasa
+  podTemplate:
+    volumes:
+    - name: githubcreds
+      secret:
+        secretName: github-token
+    - name: harborcreds
+      secret:
+        secretName: harbor-creds-yaml
+    - name: mavencreds
+      secret:
+        secretName: maven-creds
+
+EOF
+
+    output=$(kubectl -n galasa-build create -f temp/deploy-maven-galasa.yaml)
+    # Outputs a line of text like this: 
+    # pipelinerun.tekton.dev/deploy-maven-galasa-jzcvf created
+    rc=$?
+    if [[ "${rc}" != "0" ]]; then
+        error "Failed to deploy maven artifacts. rc=$?"
+        exit 1
+    fi
+    info "kubectl create pipeline run output: $output"
+
+
+    pipeline_run_name=$(echo $output | grep "created" | cut -f1 -d" " | xargs)
+
+    MAX_WAIT_ITERATIONS=30
+    COUNTER=0
+    while [  $COUNTER -lt $MAX_WAIT_ITERATIONS ]; do
+        info "Sleeping, waiting for pipeline run ${pipeline_run_name} to succeed.." 
+        # Sleep for a second.
+        sleep 10
+        let COUNTER=COUNTER+1 
+
+        # Find the status of the pipeline run...
+        status=$(kubectl get $pipeline_run_name | tail -1 | xargs | cut -f2 -d' ')
+        if [[ "${status}" == "True" ]]; then
+            info "Pipeline run completed OK."
+            break
+        fi
+
+        if [[ "${status}" == "False" ]]; then
+            error "Pipeline run $pipeline_run_name failed."
+            exit 1
+            break
+        fi
+    done
+    
+    if [ ${COUNTER} -ge $MAX_WAIT_ITERATIONS ]; then 
+        error "Timed out waiting for pipeline run ${pipeline_run_name} to complete."
+        exit 1
+    fi
+
+    success "All maven artifacts in github called ${release_type} are now deployed. Yay!"
```

Comment:
Same as my previous comment, please reflect what is actually being deployed here